src/smt/preprocess.ml

```diff
@@ -187,10 +186,6 @@ let prepreproc sq =
           let hs,_ = List.split res in
           Smt.record_signatures := hs :: !Smt.record_signatures;
           Record (List.map (fun (s, e) -> (s, self#expr scx e)) res) @@ e
-      | Apply ({core = Internal B.Eq}, [x ; {core = Internal (B.TRUE | B.FALSE)} as y])
```

Comment:
I believe you disabled this rewrite because `=' and `<=>' have different strength and so there could be an unsoundness if the rewrite was applied to formulas with unsuitable polarity? Would you mind leaving a comment in the code just as you explained the removal of the rewrite about CHOOSE?